acpi/src/lib.rs

```diff
@@ -446,3 +451,37 @@ where
         })
     }
 }
+
+pub struct SdtHeaderIterator<'t, H>
+where
+    H: AcpiHandler,
+{
+    tables_phys_ptrs: TablesPhysPtrsIter<'t>,
+    handler: H,
+}
+
+impl<'t, H> Iterator for SdtHeaderIterator<'t, H>
+where
+    H: AcpiHandler,
+{
+    type Item = SdtHeader;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        loop {
+            let table_phys_ptr = self.tables_phys_ptrs.next()?;
+            // SAFETY: `address` needs to be valid for the size of `SdtHeader`, or the ACPI tables are malformed (not a
+            // software issue).
+            let header_mapping = unsafe {
+                self.handler.map_physical_region::<SdtHeader>(table_phys_ptr as usize, mem::size_of::<SdtHeader>())
+            };
+            let r = header_mapping.validate(header_mapping.signature);
+            if r.is_err() {
+                log::warn!("Found invalid SSDT at physical address {:p}: {:?}", table_phys_ptr, r);
```

Comment:
SSDTs are a specific type of table - this should probably be `SDT` to refer to any sort of ACPI table

Reply:
Ok~ I'll fix it.

Reply:
Ok~ I'll fix it.